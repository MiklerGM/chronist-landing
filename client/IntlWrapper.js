import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';

import localeDataRU from './locales/ru.json';
// import localeDataEN from './locales/en.json';

class IntlWrapper extends React.Component {
  render() {
    return (
      <IntlProvider
        locale={this.props.locale}
        key={this.props.locale}
        messages={this.props.messages}
      >
        {this.props.children}
      </IntlProvider>
    );
  }
}

IntlWrapper.propTypes = {
  locale: PropTypes.string,
  messages: PropTypes.object
};

IntlWrapper.defaultProps = {
  locale: 'ru',
  messages: localeDataRU
};

export default IntlWrapper;
