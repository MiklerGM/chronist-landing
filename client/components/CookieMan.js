import React from 'react';
import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';

class CookieManager extends React.Component {
  state = {
    haveSession: false,
  }

  componentDidMount() {
    const token = localStorage.getItem('token');
    if (token) {
      this.setState({ haveSession: true });
    }
  }

  handleClick() {
    localStorage.setItem('token', true);
    this.setState({ haveSession: true });
  }

  render() {
    if (this.state.haveSession) {
      return null;
    }
    return (
      <div className='cookie-message'>
        <div className='page--content'>
          <p>
            <FormattedHTMLMessage id='cookie.message' />
          </p>
          <button style={{ backgroundColor: '#fafafa', color: 'black' }} onClick={() => this.handleClick()}>
            <FormattedMessage id='cookie.agree' />
          </button>
        </div>
      </div>
    );
  }
}

export default CookieManager;
