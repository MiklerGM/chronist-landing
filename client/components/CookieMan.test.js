import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import IntlWrapper from '../IntlWrapper';
import CookieMan from './CookieMan';

const TestComponent = () => (
  <IntlWrapper>
    <CookieMan />
  </IntlWrapper>
);

// it must render
it('CookieMan render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

it('CookieMan render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

// Snapshot tests
it('CookieMan Snapshot', () => {
  const tree = renderer
    .create(<TestComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
