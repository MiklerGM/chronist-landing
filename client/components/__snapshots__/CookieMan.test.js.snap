// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieMan Snapshot 1`] = `
<div
  className="cookie-message"
>
  <div
    className="page--content"
  >
    <p>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "Мы используем cookies и метрику <a href='https://yandex.ru/support/metrika/general/notification.html'>Яндекс</a> и <a href='https://policies.google.com/technologies/partner-sites?hl=ru'>Google</a>",
          }
        }
      />
    </p>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "color": "black",
        }
      }
    >
      <span>
        Согласен
      </span>
    </button>
  </div>
</div>
`;
